odxtools/cli/compare.py

```diff
@@ -0,0 +1,730 @@
+                        dl2 not in changes_variants["deleted_diagnostic_layers"]):
+
+                    changes_variants[
+                        "deleted_diagnostic_layers"].append(  # type: ignore[union-attr]
+                            dl2)
+
+                if dl1.short_name == dl2.short_name and dl1.short_name in self.diagnostic_layer_names:
+                    # compare diagnostic services of both diagnostic layers
+                    # save diagnostic service changes in dictionary (empty if no changes)
+                    service_dict: SpecsServiceDict = self.compare_diagnostic_layers(dl1, dl2)
+                    if service_dict:
+                        # adds information about diagnostic service changes to return variable (changes_variants)
+                        changes_variants.update({dl1.short_name: service_dict})
+
+        return changes_variants
+
+
+def add_subparser(subparsers: "argparse._SubParsersAction") -> None:
+    parser = subparsers.add_parser(
+        "compare",
+        description="\n".join([
+            "Compares two versions of diagnostic layers or databases with each other. Checks whether diagnostic services and its parameters have changed.",
+            "",
+            "Examples:",
+            "  Comparison of two diagnostic layers:",
+            "    odxtools compare ./path/to/database.pdx -v variant1 variant2",
+            "  Comparison of two database versions:",
+            "    odxtools compare ./path/to/database.pdx -db ./path/to/old-database.pdx",
+            "  For more information use:",
+            "    odxtools compare -h",
+        ]),
+        help="Compares two versions of diagnostic layers and/or databases with each other. Checks whether diagnostic services and its parameters have changed.",
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+    _parser_utils.add_pdx_argument(parser)
+
+    parser.add_argument(
+        "-v",
+        "--variants",
+        nargs="+",
+        metavar="VARIANT",
+        required=False,
+        default=None,
+        help="Compare specified (ecu) variants to each other.",
+    )
+
+    parser.add_argument(
+        "-db",
+        "--database",
+        nargs="+",
+        default=None,
+        metavar="DATABASE",
+        required=False,
+        help="Compare specified database file(s) to database file of first input argument.",
+    )
+
+    parser.add_argument(
+        "-nd",
+        "--no-details",
+        action="store_false",
+        default=True,
+        required=False,
+        help="Don't show all service parameter details",
+    )
+
+    parser.add_argument(
+        "-po",
+        "--plumbing-output",
+        action="store_true",
+        required=False,
+        help="Print full objects instead of selected and formatted attributes",
+    )
+    # TODO
+    # Idea: provide folder with multiple pdx files as argument
```

Comment:
maybe create an issue? otherwise this will get lost in all the code

Reply:
Ideas for improvement of comparison tool: https://github.com/mercedes-benz/odxtools/issues/252